kinesis.go

```diff
@@ -76,7 +77,7 @@ func makeParams(action string) map[string]string {
 // RequestArgs store params for request
 type RequestArgs struct {
 	params  map[string]interface{}
-	Records []PutRecordsRecord
+	Records []RecordToPut
```

Comment:
Is there a naming conflict that would stop us from using `Record`?  `Records [] of Record`

From the docs:

> PutRecord returns the shard ID of where the __data record__ was placed and the sequence number that was assigned to the data record.

Seems like `DataRecord` might fit into DSL too.